package/src/components/ImageGallery/ImageGallery.tsx

```diff
@@ -1156,30 +1160,32 @@ export const ImageGallery = <
         closeGridView={closeGridView}
         currentBottomSheetIndex={currentBottomSheetIndex}
       />
-      <BottomSheet
-        animatedIndex={animatedBottomSheetIndex}
-        containerHeight={fullScreenHeight}
-        handleComponent={() => (
-          <ImageGridHandle
+      <BottomSheetModalProvider>
+        <BottomSheetModal
+          animatedIndex={animatedBottomSheetIndex}
+          enablePanDownToClose={true}
+          handleComponent={() => (
+            <ImageGridHandle
+              closeGridView={closeGridView}
+              {...imageGalleryCustomComponents?.gridHandle}
+            />
+          )}
+          handleHeight={imageGalleryGridHandleHeight ?? 40}
+          index={0}
+          onChange={(index: number) => setCurrentBottomSheetIndex(index)}
+          ref={bottomSheetModalRef}
+          snapPoints={imageGalleryGridSnapPoints || snapPoints}
+        >
+          <ImageGrid
             closeGridView={closeGridView}
-            {...imageGalleryCustomComponents?.gridHandle}
+            numberOfImageGalleryGridColumns={numberOfImageGalleryGridColumns}
+            photos={photos}
+            resetVisibleValues={resetVisibleValues}
+            setImage={setImage}
+            {...imageGalleryCustomComponents?.grid}
           />
```

Comment:
Do you think it'd be possible to extract these to a separate smaller component, or do you it'd make things trickier? Would be nice to break down the bigger components we have when we have the chance I think, but there might be things I haven't considered for this case

Reply:
I think we can take up that as a separate PR(if it doesn't seem to be a blocker) because currently it doesn't seem to be doable in one go and might need some head-up configuring the code. I will create a separate task on JIRA for this if you are fine with it?

Reply:
Sure, if you think that's the better option, I'm happy to do it that way. I can also share a diff of what I'm thinking of if you'd like/think that might be of value